src/main/resources/modules/hiv_diagnosis.json

```diff
@@ -0,0 +1,273 @@
+      },
+      "direct_transition": "Terminal",
+      "referenced_by_attribute": "hiv_diagnosis"
+    },
+    "Diagnosis_Encounter_Later_Years": {
+      "type": "Encounter",
+      "encounter_class": "ambulatory",
+      "codes": [
+        {
+          "system": "SNOMED-CT",
+          "code": 185347001,
+          "display": "Encounter for problem (procedure)"
+        }
+      ],
+      "direct_transition": "Screening Submodule"
+    },
+    "Diagnosis_Encounter_Early_Years": {
+      "type": "Encounter",
+      "encounter_class": "ambulatory",
+      "codes": [
+        {
+          "system": "SNOMED-CT",
+          "code": 185347001,
+          "display": "Encounter for problem (procedure)"
+        }
+      ],
+      "direct_transition": "Screening_Submodule"
+    },
+    "Diagnosis_Encounter_End": {
+      "type": "EncounterEnd",
+      "direct_transition": "Mortality Check Later Years"
+    },
+    "Diagnosis_Encounter_End_Early_Years": {
+      "type": "EncounterEnd",
+      "direct_transition": "Mortality Check"
+    },
+    "Screening Submodule": {
+      "type": "CallSubmodule",
+      "submodule": "hiv/hiv_screening",
+      "direct_transition": "Diagnosis Later Years"
+    },
+    "Screening_Submodule": {
+      "type": "CallSubmodule",
+      "submodule": "hiv/hiv_screening",
+      "direct_transition": "Diagnosis Early Years"
+    },
+    "Diagnosed_Later_Years": {
+      "type": "SetAttribute",
+      "attribute": "hiv_infection",
+      "direct_transition": "Diagnosis_Encounter_Later_Years",
+      "value": true
+    },
+    "Diagnosed_Early_Years": {
+      "type": "SetAttribute",
+      "attribute": "hiv_infection",
+      "direct_transition": "Diagnosis_Encounter_Early_Years",
+      "value": true
+    },
+    "Wait_Until_Next_Diagnosis_Check": {
+      "type": "Delay",
+      "distribution": {
+        "kind": "EXACT",
+        "parameters": {
+          "value": 1
+        }
+      },
+      "unit": "years",
+      "direct_transition": "Diagnosis Check"
+    },
+    "Diagnosis Check Early Years": {
+      "type": "Simple",
+      "lookup_table_transition": [
+        {
+          "transition": "Diagnosed_Early_Years",
+          "default_probability": 0.0001565,
+          "lookup_table_name": "hiv_diagnosis_early.csv"
+        },
+        {
+          "transition": "Wait_Until_Next_Diagnosis_Check",
+          "default_probability": 0.9998435,
+          "lookup_table_name": "hiv_diagnosis_early.csv"
+        }
+      ]
+    },
+    "Diagnosis Check Later Years": {
+      "type": "Simple",
+      "lookup_table_transition": [
+        {
+          "transition": "Diagnosed_Later_Years",
+          "default_probability": 0.000097,
+          "lookup_table_name": "hiv_diagnosis_later.csv"
+        },
+        {
+          "transition": "Wait_Until_Next_Diagnosis_Check",
+          "default_probability": 0.999903,
+          "lookup_table_name": "hiv_diagnosis_later.csv"
+        }
+      ]
+    },
+    "Diagnosis Check": {
```

Comment:
I think we need a Guard state that will wait until 1979. My reading of this is that if anyone is born prior to 1979, when this runs, it will just go to terminal. I ran 10K patients and none of those diagnosed with HIV was born before 1979.

Reply:
I've made the change, now I'm testing it and running the data analysis.

Reply:
Guard added and checked in. Resulting graphs from below.